src/live/src/user-to-user/stream.js

```diff
@@ -0,0 +1,245 @@
+   * Unsubscribes the active user from the specified user's channel
+   * @param {string} userId
+   * @returns {Promise}
+   */
+  unfollow(userId) {
+    return this._unsubscribe(userId);
+  }
+
+  /**
+   * Posts a message to the active user's channel
+   * @param {Object} message
+   * @returns {Promise}
+   */
+  post(message) {
+    const userId = this.client.activeUser._id;
+    return this._publish(userId, message);
+  }
+
+  // Directed comm
+
+  /**
+   * Listens for messages sent to the active user
+   * @param {MessageReceiver} receiver
+   */
+  listen(receiver) {
+    const userId = this.client.activeUser._id;
+    return this._subscribe(userId, receiver);
+  }
+
+  /**
+   * Stops listening for messages sent to the active user
+   */
+  stopListening() {
+    const userId = this.client.activeUser._id;
+    return this._unsubscribe(userId);
+  }
+
+  /**
+   * Sends a message to the specified user
+   * @param {string} userId
+   * @param {Object} message
+   */
+  send(userId, message) {
+    return this._publish(userId, message);
+  }
+
+  /**
+   * @private
+   * @param {string} substreamOwnerId
+   * @returns {Promise}
+   */
+  _unsubscribeFromSubstream(substreamOwnerId) {
+    const path = `${substreamOwnerId}/unsubscribe`;
+
+    return this._makeStreamRequest(path, RequestMethod.POST, { deviceId: this.client.deviceId })
+      .then((response) => {
+        delete this.subscribeChannels[substreamOwnerId];
+        return response;
+      });
+  }
+
+  /**
+   * @private
+   * @param {string} userId
+   * @param {MessageReceiver} receiver
+   * @returns {Promise}
+   */
+  _subscribe(userId, receiver) {
+    return this._requestSubscribeAccess(userId)
+      .then(() => {
+        const channelName = this.subscribeChannels[userId] || this._buildChannelName(userId);
+        return this.liveService.subscribeToChannel(channelName, receiver);
+      });
+  }
+
+  /**
+   * @private
+   * @param {string} userId
+   * @returns {Promise<void>}
+   */
+  _unsubscribe(userId) {
+    return this._unsubscribeFromSubstream(userId)
+      .then(() => {
+        const channelName = this.subscribeChannels[userId] || this._buildChannelName(userId);
+        this.liveService.unsubscribeFromChannel(channelName);
+      });
+  }
+
+  /**
+   * @private
+   * @param {string} userId
+   * @param {Object} message
+   * @returns {Promise}
+   */
+  _publish(userId, message) {
+    if (this.publishChannels[userId]) {
+      return this._retriablePublish(userId, message);
+    }
+    return this._publishWithAccessRequest(userId, message);
+  }
+
+  /**
+   * @private
+   * @param {string} userId
+   * @param {Object} message
+   * @returns {Promise}
+   */
+  _publishWithAccessRequest(userId, message) {
+    return this._requestPublishAccess(userId)
+      .then(() => {
+        const channelName = this.publishChannels[userId] || this._buildChannelName(userId);
+        return this.liveService.publishToChannel(channelName, message);
+      });
```

Comment:
What happens if the initial `_requestPublishAccess` returns a `StatusCode.Forbidden` because the user does not have publish rights?  Does this need to be handled here?